src/asl.ts

```diff
@@ -381,6 +381,8 @@ export class ASL {
   private readonly stateNamesCount = new Map<string, number>();
   private readonly generatedNames = new DeterministicNameGenerator();
 
+  private needsFunctionlessContext: boolean = false;
```

Comment:
comment to add some context on what this is used for? Define functionless context - I assume this is for lexical scoping and constants?